activesupport/lib/active_support/core_ext/module/attribute_accessors.rb

```diff
@@ -38,22 +38,20 @@ class Module
   #
   #   Person.new.hair_colors # => NoMethodError
   #
-  #
-  # Also, you can pass a block to set up the attribute with a default value.
+  # You can set a default value for the attribute.
   #
   #   module HairColors
-  #     mattr_reader :hair_colors do
-  #       [:brown, :black, :blonde, :red]
-  #     end
+  #     mattr_reader :hair_colors, default: [:brown, :black, :blonde, :red]
   #   end
   #
   #   class Person
   #     include HairColors
   #   end
   #
   #   Person.new.hair_colors # => [:brown, :black, :blonde, :red]
-  def mattr_reader(*syms)
-    options = syms.extract_options!
+  def mattr_reader(*syms, instance_reader: true, instance_accessor: true, default: nil)
+    default = yield if block_given? && default.nil?
```

Comment:
Moving the `yield` out of the loop is a behavioural change.. probably better avoided.